OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -280,11 +280,22 @@ public override void Cancel(Actor self)
 
 		public override IEnumerable<Target> GetTargets(Actor self)
 		{
-			if (path != null)
-				return Enumerable.Reverse(path).Select(c => Target.FromCell(self.World, c));
-			if (destination != null)
-				return new Target[] { Target.FromCell(self.World, destination.Value) };
-			return Target.None;
+			var validTargets = new List<Target>();
+
+			var activityIterator = (Activity)this;
+			while (activityIterator != null)
+			{
+				if (!activityIterator.IsCanceled && activityIterator is Move)
```

Comment:
Err, hang on... You really don't want to be walking the activity queue from *inside* an activity on the queue!